datasets/rvl_cdip/rvl_cdip.py

```diff
@@ -0,0 +1,160 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""RVL-CDIP (Ryerson Vision Lab Complex Document Information Processing) dataset"""
+
+
+import os
+
+import datasets
+from datasets.tasks import ImageClassification
+
+
+_CITATION = """\
+@inproceedings{harley2015icdar,
+    title = {Evaluation of Deep Convolutional Nets for Document Image Classification and Retrieval},
+    author = {Adam W Harley and Alex Ufkes and Konstantinos G Derpanis},
+    booktitle = {International Conference on Document Analysis and Recognition ({ICDAR})}},
+    year = {2015}
+}
+"""
+
+
+_DESCRIPTION = """\
+The RVL-CDIP (Ryerson Vision Lab Complex Document Information Processing) dataset consists of 400,000 grayscale images in 16 classes, with 25,000 images per class. There are 320,000 training images, 40,000 validation images, and 40,000 test images.
+"""
+
+
+_HOMEPAGE = "https://www.cs.cmu.edu/~aharley/rvl-cdip/"
+
+
+_LICENSE = "https://www.industrydocuments.ucsf.edu/help/copyright/"
+
+
+_URLS = {
+    "rvl-cdip": "https://huggingface.co/datasets/mariosasko/rvl_cdip/resolve/main/rvl-cdip.tar.gz",
+}
+
+_METADATA_URLS = {
+    "train": "https://huggingface.co/datasets/mariosasko/rvl_cdip/resolve/main/train.txt",
+    "test": "https://huggingface.co/datasets/mariosasko/rvl_cdip/resolve/main/test.txt",
+    "val": "https://huggingface.co/datasets/mariosasko/rvl_cdip/resolve/main/val.txt",
+}
+
+_CLASSES = [
+    "letter",
+    "form",
+    "email",
+    "handwritten",
+    "advertisement",
+    "scientific report",
+    "scientific publication",
+    "specification",
+    "file folder",
+    "news article",
+    "budget",
+    "invoice",
+    "presentation",
+    "questionnaire",
+    "resume",
+    "memo",
+]
+
+_IMAGES_DIR = "images/"
+
+
+class RvlCdip(datasets.GeneratorBasedBuilder):
+    """Ryerson Vision Lab Complex Document Information Processing dataset."""
+
+    VERSION = datasets.Version("1.0.0")
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "image": datasets.Image(),
+                    "label": datasets.ClassLabel(names=_CLASSES),
+                }
+            ),
+            supervised_keys=("image", "label"),
+            homepage=_HOMEPAGE,
+            citation=_CITATION,
+            license=_LICENSE,
+            task_templates=[ImageClassification(image_column="image", label_column="label")],
+        )
+
+    def _split_generators(self, dl_manager):
+        archive_path = dl_manager.download(_URLS["rvl-cdip"])
+        labels_path = dl_manager.download(_METADATA_URLS)
+
+        # (Optional) In non-streaming mode, we can extract the archive locally to have actual local image files:
+        local_extracted_archive = dl_manager.extract(archive_path) if not dl_manager.is_streaming else None
```

Comment:
I'm not a big fan of doing this, why do we need to extract it locally ? Can't we just use the extracted data from `dl_manager.iter_archive` ?

(for the record, we've added this feature for audio datasets because some audio formats can't easily be decoded from a fileobj and need actual files on disk - this is not the cause for images AFAIK)

Reply:
Got it. This makes sense. I'll remove the extract part. :+1: